lms/djangoapps/instructor_analytics/basic.py

```diff
@@ -340,3 +345,63 @@ def dump_grading_context(course):
     msg += "length=%d\n" % len(gcontext['all_descriptors'])
     msg = '<pre>%s</pre>' % msg.replace('<', '&lt;')
     return msg
+
+
+def iterate_problem_components(course):
+    """ Helper function for iterating through all problems of a course """
+    for section in course.get_children():
+        section_name = section.display_name_with_default
+        for subsection in section.get_children():
+            subsection_name = subsection.display_name_with_default
+            for unit in subsection.get_children():
+                unit_name = unit.display_name_with_default
+
+                parent_metadata = [section_name, subsection_name, unit_name]
+                for component in unit.get_children():
+                    if component.category == 'problem':
+                        yield component, parent_metadata
+
+
+def student_responses(course):
+    """
+    Yields student responses for all problems in course for writing out to a CSV file.
+    """
+    order = 1
```

Comment:
This is for a column in the CSV for problem order in course, so instructors could sort by it.